### PR TITLE
Adds google analytics to content security policy

### PR DESCRIPTION
**Pull request by seanfeld**
# Google analytics content security policy

## Description
Adds Tag manager and analytics to the content security policy based on the guidance here https://developers.google.com/tag-platform/tag-manager/web/csp .

## PR Type
- [x] Bug Fix
- [ ] Feature
- [ ] Build Fix
- [ ] Testing
- [ ] General Improvement
- [ ] Cherry Pick

### CODE REVIEW

File: server/express.js [hunk 1]
```diff
@@ -157,14 +158,23 @@ function makeApp(authAddress, cdapConfig, uiSettings) {
       helmet({
         contentSecurityPolicy: {
           directives: {
-            imgSrc: [`'self' data: ${cspWhiteListUrls}`],
+            imgSrc: [`'self' data: ${cspWhiteListUrls}`, 'www.googletagmanager.com', 'https://www.gstatic.com', 'https://ssl.gstatic.com', 'https://www.google-analytics.com'],
```
Comment (njbriggs) [inline server/express.js:162]: Why do some domains have a scheme (`https`) but not others?

Comment (seanfeld) [reply server/express.js:162]: the image source one is recommended that way in the doc on the analytics site

**Review by sumengwang — APPROVED** [review]

File: server/express.js [hunk 2]
```diff
@@ -85,6 +85,7 @@ export const stripAuthHeadersInProxyMode = (cdapConfig, res) => {
 
 function makeApp(authAddress, cdapConfig, uiSettings) {
   var app = express();
+  const serverNonce = Math.random().toFixed(16).split('.')[1];
```
Comment (dli357) [inline server/express.js]: The server nonce should not be guessable as mentioned by https://developers.google.com/tag-platform/tag-manager/web/csp, and `Math.random()` is not considered a cryptographically secure random number generator. Can we switch to using a cryptographically secure PRNG?

**Review by dli357 — CHANGES_REQUESTED** [review]

File: server/express.js [hunk 3]
```diff
@@ -85,6 +86,7 @@ export const stripAuthHeadersInProxyMode = (cdapConfig, res) => {
 
 function makeApp(authAddress, cdapConfig, uiSettings) {
   var app = express();
+  const serverNonce = crypto.randomBytes(16).toString('base64');
```
Comment (seanfeld) [inline server/express.js]: @dli357

File: server/express.js [hunk 4]
```diff
@@ -157,14 +159,23 @@ function makeApp(authAddress, cdapConfig, uiSettings) {
       helmet({
         contentSecurityPolicy: {
           directives: {
-            imgSrc: [`'self' data: ${cspWhiteListUrls}`],
+            imgSrc: [`'self' data: ${cspWhiteListUrls}`, 'www.googletagmanager.com', 'https://www.gstatic.com', 'https://ssl.gstatic.com', 'https://www.google-analytics.com'],
+            connectSrc: [`'self'`, 'https://www.google-analytics.com', 'https://ampcid.google.com'],
+            fontSrc: [`'self'`, `'unsafe-inline'`, 'https://fonts.gstatic.com', 'data:'],
             scriptSrc: [
               (req, res) => `'nonce-${res.locals.nonce}'`,
               `'unsafe-inline'`,
```
Comment (dli357) [inline server/express.js:167]: nit: Can we open a JIRA to track an improvement to remove `unsafe-inline` and `unsafe-eval` down the road?

**Review by dli357 — APPROVED** [review]

File: server/express.js (hunk 4 above)
Comment (seanfeld) [reply server/express.js:167]: @njbriggs https://cdap.atlassian.net/browse/CDAP-20666

**Pull request merged**